internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -967,44 +975,45 @@ func (s *Server) processPullRequestEvent(
 		return nil, err
 	}
 
-	provider, err := s.providerManager.InstantiateFromID(ctx, repoEnt.Entity.ProviderID)
+	pullProps, err := properties.NewProperties(map[string]any{
+		properties.PropertyUpstreamID: event.GetPullRequest().GetID(),
+		ghprop.PullPropertyRepoName:   ghRepo.GetName(),
+		ghprop.PullPropertyRepoOwner:  ghRepo.GetOwner(),
+		ghprop.PullPropertyURL:        event.GetPullRequest().GetURL(),
+		ghprop.PullPropertyNumber:     event.GetPullRequest().GetNumber(),
+		ghprop.PullPropertyAuthorID:   event.GetPullRequest().GetUser().GetID(),
+		ghprop.PullPropertyAction:     event.GetAction(),
+	})
 	if err != nil {
-		l.Error().Err(err).Msg("error instantiating provider")
-		return nil, err
+		return nil, fmt.Errorf("error creating pull request properties: %w", err)
 	}
 
-	cli, err := provifv1.As[provifv1.GitHub](provider)
```

Comment:
We just treat the provider as a generic provider here, not as github, except creating the properties to look up by and then calling the GH-specific function to format the fetched properties to pb.PullRequest